api/v1/composition.go

```diff
@@ -214,3 +214,32 @@ func (s *CompositionStatus) GetCurrentSynthesisUUID() string {
 func (c *Composition) ShouldIgnoreSideEffects() bool {
 	return c.Annotations["eno.azure.io/ignore-side-effects"] == "true"
 }
+
+func (c *Composition) Synthesizing() bool {
+	return c.Status.CurrentSynthesis != nil && c.Status.CurrentSynthesis.Synthesized == nil
+}
+
+func (c *Composition) EnableIgnoreSideEffects() {
+	anno := c.GetAnnotations()
+	if anno == nil {
+		anno = map[string]string{}
+	}
+	anno["eno.azure.io/ignore-side-effects"] = "true"
+	c.SetAnnotations(anno)
+}
+
+const forceResynthesisAnnotation = "eno.azure.io/force-resynthesis"
+
+func (c *Composition) ForceResynthesis() {
```

Comment:
Does this mean that we can only fore resynthesis once?

Reply:
Only one synthesis UUID can be eligible for resynthesis at once, which I think makes sense?

Reply:
I don't think I saw this code, but does the annotation get cleared when the current synthesis succeeds? I ask this mainly because the self healing controller relies on being able to force a resynthesis right? I kinda dislike the idea of the self-healing controller to be honest, it's just a manifestation of races/inconsistencies somewhere else. If we still need it though, I want to make sure that it's still supported

Reply:
The annotation doesn't get cleared, it just becomes a no-op since once it's accepted the UUID will no longer match.

Spoiler alert on the change I'm working on next: the self healing controller gets merged into a new more generic "resource slice controller" along with a couple of others 😊

Reply:
That's perfect, lgtm then!